tubesync/sync/models.py

```diff
@@ -482,6 +489,15 @@ def type_directory_path(self):
     def make_directory(self):
         return os.makedirs(self.directory_path, exist_ok=True)
 
+    @property
+    def get_thumbnail_url(self):
+        if self.source_type==Source.SOURCE_TYPE_YOUTUBE_PLAYLIST:
+            raise Exception('This source is a playlist so it doesn\'t have thumbnail.')
+        soup = BeautifulSoup(requests.get(self.url, cookies={'CONSENT': 'YES+1'}).text, "html.parser")
+        data = re.search(r"var ytInitialData = ({.*});", str(soup.prettify())).group(1)
+        json_data = json.loads(data)
```

Comment:
And wrap this to catch ValueError and TypeError in case the JSON found is invalid.